### DISCUSSION TITLE
Are dependencies versioned, and if not how does straight satisfy them?

### DISCUSSION THREAD

**Discussion by dabrahams** [raxod502/straight.el#1135] [Q&A] [answered]
The README says:

> the only information straight.el gets about a package's dependencies are their names.

If that's so, how does straight satisfy a declared dependency?  Does it just check out the HEAD of `main`?  The latest release? something else?

**Comment by progfolio** [comment]
>  how does straight satisfy a declared dependency?

Straight checks the dependent package's metadata for the Package-Requires line.
It only considers the ID of each package. e.g. 

```emacs-lisp
;; Package-Requires: ((org "9.7"))
```

will tell Straight to install Org unless the user has already declared it in their init file.
That is why it's suggested to hoist certain popular dependencies above other declarations if you plan to explicitly install a version (Org falls into this camp).

No consideration is made of the version number in the metadata.
So, in our example, you would get the latest revision of Org when it's installed as a dependency of another package.

A feature we are considering is to at least issue warnings when a dependency's minimum version as required by its dependent is not satisfied. You can follow that discussion here:

https://github.com/radian-software/straight.el/issues/1083

Does that help?

**Reply by dabrahams** [reply]
Almost.  What do you mean, exactly, by "latest revision?"  Is it the latest release?  The latest commit at the tip of the default branch?

**Reply by raxod502** [reply]
The latest commit. There is no standard release schema for Emacs packages that could be determined by looking at the Git repository.

**Reply by dabrahams** [reply]
Sorry, the latest commit at the tip of what branch?

**Reply by progfolio** [reply, marked as answer]
That's the branch the remote's HEAD points to. e.g.

```shell
$ git ls-remote --symref "https://git.savannah.gnu.org/git/emacs/org-mode.git" HEAD
ref: refs/heads/main    HEAD
de9f1169d2293ac4c7d554196ba31ff47daecbe0        HEAD

# VS

$ git ls-remote --symref "https://git.savannah.gnu.org/git/emacs.git" HEAD
ref: refs/heads/master  HEAD
a1ac7d0720d59aee3534dc518abccf1a77cd5e5a        HEAD
```
